modules/iam-assumable-role-with-oidc/main.tf

```
@@ -96,7 +96,7 @@ resource "aws_iam_role" "this" {
   force_detach_policies = var.force_detach_policies
   permissions_boundary  = var.role_permissions_boundary_arn
 
-  assume_role_policy = data.aws_iam_policy_document.assume_role_with_oidc[0].json
+  assume_role_policy = var.create_custom_role_trust_policy ? var.custom_role_trust_policy : data.aws_iam_policy_document.assume_role_with_oidc[0].json
```

I believe we can simply use:
```suggestion
  assume_role_policy = coalesce(var.assume_role_policy, data.aws_iam_policy_document.assume_role_with_oidc[0].json)
```

---

the other changes can be carried throughout the other modules